src/ast/rewriter/seq_rewriter.cpp

```diff
@@ -3339,6 +3339,26 @@ expr_ref seq_rewriter::mk_derivative_rec(expr* ele, expr* r) {
             //while mk_empty() = [], because deriv(epsilon) = [] = nothing
             return mk_empty();
         }
+        else if (str().is_itos(r1, r2))
```

Comment:
I don't see how it fixes the underlying issue. The default behavior is to return a derivative expression. It should be sound no matter what it is applied to. Adding this code just bypasses the default case so it seems to mask the bug, not outright address it for other scenarios. Suppose I introduce an uninterpreted function that has the same effect as itos. This code would go to the default case.